src/messaging/ExchangeMgr.cpp

```diff
@@ -94,6 +98,10 @@ CHIP_ERROR ExchangeManager::Init(SecureSessionMgr * sessionMgr)
 
 CHIP_ERROR ExchangeManager::Shutdown()
 {
+// TODO(#6931): Lock guard is a temporary solution. Proper solution should be post chip shutdown function into chip thread
+#if CONFIG_DEVICE_LAYER
+    DeviceLayer::PlatformMgr().LockChipStack();
```

Comment:
I would suggest moving DeviceLayer::PlatformMgr().LockChipStack() to the place where shutdown is called from the client thread, instead of within ExchangeManager::Shutdown() itself.

You are trying to fix the problem in the case which Shutdown is called from a client thread, for the cases Shutdown() are called from CHIP main thread event loop(Which it should be), you will be in a deadlock.